### PR TITLE
Add Package HSF-CMakeTools

### PR DESCRIPTION
**Pull request by vvolkl**

### CODE REVIEW

File: var/spack/repos/builtin/packages/hsf-cmaketools/package.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    url      = "https://github.com/HSF/cmaketools/archive/1.8.tar.gz"
+    git      = "https://github.com/HSF/cmaketools.git"
+
+    maintainers = ['vvolkl']
+
+    version('master', branch='master')
+    version('1.8', sha256='91af30f5701dadf80a5d7e0d808c224c934f0784a3aff2d3b69aff24f7e1db41')
+
+    def install(self, spec, prefix):
+        mkdir(join_path(prefix, "modules"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hsf-cmaketools/package.py]:
```suggestion
        mkdir(prefix.modules)
```

File: var/spack/repos/builtin/packages/hsf-cmaketools/package.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        copy_tree('modules', join_path(prefix, 'modules'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hsf-cmaketools/package.py]:
```suggestion
        install_tree('modules', prefix.modules)
```

File: var/spack/repos/builtin/packages/hsf-cmaketools/package.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        copy("CMakeToolsConfig.cmake", prefix)
+
+    def setup_dependent_build_environment(self, env, dependent_spec):
+        env.set('CMAKE_PREFIX_PATH', self.prefix)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hsf-cmaketools/package.py]: I believe this is already set in `build_environment.py`, so you shouldn't need this.

File: var/spack/repos/builtin/packages/hsf-cmaketools/package.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+    maintainers = ['vvolkl']
+
+    version('master', branch='master')
+    version('1.8', sha256='91af30f5701dadf80a5d7e0d808c224c934f0784a3aff2d3b69aff24f7e1db41')
+
+    def install(self, spec, prefix):
+        mkdir(join_path(prefix, "modules"))
+        copy_tree('modules', join_path(prefix, 'modules'))
+        copy("CMakeToolsConfig.cmake", prefix)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hsf-cmaketools/package.py]:
```suggestion
        install("CMakeToolsConfig.cmake", prefix)
```

File: var/spack/repos/builtin/packages/hsf-cmaketools/package.py (hunk 3 above)
Comment (vvolkl) [reply var/spack/repos/builtin/packages/hsf-cmaketools/package.py]: That's true, I just checked. Thanks for the comments!

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**